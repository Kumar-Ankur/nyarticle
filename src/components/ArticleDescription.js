import { Card, CardMedia, Chip, Divider, Typography } from '@mui/material';
import { useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';
import constant from '../constants/Index';
import useFetch from '../customHooks/useFetch';

const ArticleDescription = () => {
  const { id } = useParams();
  useFetch();
  const { articlesData } = useSelector((state) => state.article);

  const selectedArticle = articlesData.filter(
    (article) => article.id === Number(id),
  )[0];

  const renderFacet = (facets) => {
    return facets.map((facet, index) => {
      return <Chip key={index} label={facet}></Chip>;
    });
  };

  return (
    <div className="article_description_container">
      <Typography
        variant="h2"
        component="div"
        className="article_description_container-title"
        role="heading"
      >
        {selectedArticle?.title}
      </Typography>

      <Typography
        variant="h5"
        className="article_description_container-abstract"
      >
        {selectedArticle?.abstract}
      </Typography>

      <Typography variant="h6" className="article_description_container-byline">
        {selectedArticle?.byline}
      </Typography>

      <Typography variant="h6" className="article_description_container-byline">
        Updated: {new Date(selectedArticle?.updated).toLocaleDateString()}
      </Typography>

      <Typography variant="h6" className="article_description_container-byline">
        Published:{' '}
        {new Date(selectedArticle?.published_date).toLocaleDateString()}
      </Typography>

      <Divider style={{ width: '100%' }} />

      <Card style={{ width: '100%' }}>
        <CardMedia
          component="img"
          style={{ width: '100%', objectFit: 'contain' }}
          image={
            selectedArticle?.media?.[0]?.['media-metadata']?.[2]?.url ??
            constant.IMAGE_PLACEHOLDER_URL
          }
          title={selectedArticle?.title}
        ></CardMedia>
      </Card>

      <Divider style={{ width: '100%' }} />

      <div
        style={{
          display: 'flex',
          justifyContent: 'flex-start',
          alignItems: 'center',
          gap: '12px',
        }}
      >
        {selectedArticle?.des_facet.length &&
          renderFacet(selectedArticle.des_facet)}
      </div>
    </div>
  );
};

export default ArticleDescription;
