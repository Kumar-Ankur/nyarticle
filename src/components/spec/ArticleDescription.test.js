import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import ArticleDescription from '../ArticleDescription';
import { BrowserRouter } from 'react-router-dom';
import { useSelector } from 'react-redux';
// import useFetch from '../../customHooks/useFetch';
// useFetch

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}));

jest.mock('../../customHooks/useFetch')

describe('ArticleDescription component',  () => {
  test('renders article description correctly', () => {
    // Mocked article data
    const articleData = {
      id: 1,
      title: 'Test Article',
      abstract: 'Test abstract',
      byline: 'Test byline',
      updated: '2022-01-01',
      published_date: '2022-01-01',
      media: [
        {
          'media-metadata': [{ url: 'test_image_url' }],
        },
      ],
      des_facet: ['facet1', 'facet2'],
    };

    // Mock useSelector hook return value
    useSelector.mockReturnValue({
       articlesData: [articleData] ,
    });

    const { asFragment }= render(
      <BrowserRouter>
        <ArticleDescription />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
