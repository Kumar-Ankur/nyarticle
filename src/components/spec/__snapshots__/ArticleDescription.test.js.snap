// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleDescription component renders article description correctly 1`] = `
<DocumentFragment>
  <div
    class="article_description_container"
  >
    <div
      class="MuiTypography-root MuiTypography-h2 article_description_container-title css-1sra7t5-MuiTypography-root"
      role="heading"
    />
    <h5
      class="MuiTypography-root MuiTypography-h5 article_description_container-abstract css-ag7rrr-MuiTypography-root"
    />
    <h6
      class="MuiTypography-root MuiTypography-h6 article_description_container-byline css-2ulfj5-MuiTypography-root"
    />
    <h6
      class="MuiTypography-root MuiTypography-h6 article_description_container-byline css-2ulfj5-MuiTypography-root"
    >
      Updated: Invalid Date
    </h6>
    <h6
      class="MuiTypography-root MuiTypography-h6 article_description_container-byline css-2ulfj5-MuiTypography-root"
    >
      Published: Invalid Date
    </h6>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      style="width: 100%;"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      style="width: 100%;"
    >
      <img
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
        src="https://s3-us-west-2.amazonaws.com/s.cdpn.io/1255/image-not-found.svg"
        style="width: 100%; object-fit: contain;"
      />
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      style="width: 100%;"
    />
    <div
      style="display: flex; justify-content: flex-start; align-items: center; gap: 12px;"
    />
  </div>
</DocumentFragment>
`;
